niceplots/utils.py

```diff
@@ -197,6 +197,59 @@ def draggable_legend(axis=None, color_on=True, **kwargs):
         legend[idx].draggable()
 
 
+def label_line_ends(ax, lines=None, labels=None, colors=None, **kwargs):
+    """Place a label just to the right of each line in the axes
+
+    Note: Because the labels are placed outside of the axes, this function works best for plots where all lines end as
+    close to the right edge of the axes as possible. Additionally you need to either use constrained_layout=True
+    (as NicePlots styles do), or call plt.tight_layout() after calling this function.
+
+    Parameters
+    ----------
+    ax : Matplotlib axes
+        axes to label the lines of
+    lines : iterable of matplotlib line objects, optional
+        Lines to label, by default all lines in the axes
+    labels : list of strings, optional
+        Labels for each line, by default uses each line's label attribute
+    colors : single or list of colors, optional
+        Color(s) to use for each line, can be a single color for all lines or a list containing an entry for each line,
+        by default uses each line's color
+    **kwargs :
+        Any valid keywords for matplotlib's annotate function
+    """
+    if lines is None:
+        lines = ax.get_lines()
+
+    if labels is None:
+        labels = [line.get_label() for line in lines]
+    numLines = len(lines)
+    if len(labels) != numLines:
+        raise ValueError(f"Number of labels ({len(labels)}) doesn't match number of lines ({numLines})")
+
+    if colors is None:
+        colors = [line.get_color() for line in lines]
+    if not isinstance(colors, list):
+        colors = [colors] * len(lines)
+    if len(colors) != numLines:
+        raise ValueError(f"Number of colors ({len(colors)}) doesn't match number of lines ({numLines})")
+
+    for line, label, color in zip(lines, labels, colors):
+        # Get the x, y coordinates of the right-most point on the line
+        maxXIndex = np.argmax(line.get_xdata())
+        x = line.get_xdata()[maxXIndex]
+        y = line.get_ydata()[maxXIndex]
+        ax.annotate(
+            label,
+            xy=(x, y),
+            xytext=(6, 0),
```

Comment:
We may have already discussed this, but what do you think about making the x offset (and perhaps y offset) a single scalar optional argument to `label_line_ends` so that it the spacing can be adjusted if the user makes the figure size huge or tiny?

Reply:
Done